web/ui/src/Services/Service.js

```diff
@@ -0,0 +1,361 @@
+            this.fetchMonitoringProfile(force);
+            this.fetchExportEndpoints(force);
+        }
+
+        fetchAddresses(force) {
+            fetch.call(this, "getServiceIpAssignments", "addresses", force);
+        }
+
+        fetchConfigs(force) {
+            fetch.call(this, "getServiceConfigs", "configs", force);
+        }
+
+        fetchEndpoints(force) {
+            // populate publicEndpoints property
+            fetch.call(this, "getServicePublicEndpoints", "publicEndpoints", force);
+        }
+
+        fetchExportEndpoints(force) {
+            let deferred = $q.defer();
+            resourcesFactory.v2.getServiceExportEndpoints(this.id)
+                .then(response => {
+                    this.exportedServiceEndpoints = response
+                        .filter(ept => ept.Protocol === "tcp")
+                        .map(ept => {
+                            // TODO - dont modify model data :(
+                            ept.Value = `${ept.ServiceName} - ${ept.Application}`;
+                            return ept;
+                        });
+                    deferred.resolve();
+                },
+                error => {
+                    console.warn(error);
+                    deferred.reject();
+                });
+
+            return deferred.promise;
+        }
+
+        fetchMonitoringProfile(force) {
+            fetch.call(this, "getServiceMonitoringProfile", "monitoringProfile", force);
+        }
+
+        fetchInstances() {
+            let deferred = $q.defer();
+            resourcesFactory.v2.getServiceInstances(this.id)
+                .then(data => {
+                    // TODO - dont blow away existing instances
+                    this.instances = data.map(i => new Instance(i));
+                    deferred.resolve();
+                },
+                error => {
+                    console.warn(error);
+                    deferred.reject();
+                });
+
+            return deferred.promise;
+        }
+
+        fetchServiceChildren(force) {
+            let deferred = $q.defer();
+            if (this.subservices.length && !force) {
+                deferred.resolve();
+            }
+            resourcesFactory.v2.getServiceChildren(this.id)
+                .then(data => {
+                    // TODO - dont blow away existing children
+                    this.subservices = data.map(s => new Service(s));
+                    this.touch();
+                    deferred.resolve();
+                },
+                error => {
+                    console.warn(error);
+                    deferred.reject();
+                });
+            return deferred.promise;
+        }
+
+        // fast-moving state for this service and its instances
+        // note: returns a promise that resolves with a single status object
+        getStatus() {
+            let deferred = $q.defer();
+
+            resourcesFactory.v2.getServiceStatus(this.id)
```

Comment:
This is for the current service only -- the one mentioned at the top of the page - so we can fill in all the instance tables and endpoints etc. We batch lite service status calls for all the sub-services into a single call elsewhere.